bigtable-dataflow-parent/bigtable-beam-import/src/test/java/com/google/cloud/bigtable/beam/test_env/TestProperties.java

```diff
@@ -0,0 +1,155 @@
+import org.apache.beam.sdk.extensions.gcp.options.GcpOptions;
+import org.apache.hadoop.conf.Configuration;
+
+public class TestProperties {
+  private final String projectId;
+  private final String instanceId;
+  private final String dataflowRegion;
+  private final String workdir;
+  private final String cloudDataDir;
+  private final Optional<String> dataEndpoint;
+  private final Optional<String> adminEndpoint;
+
+  public TestProperties(
+      String projectId,
+      String instanceId,
+      String dataflowRegion,
+      String workdir,
+      String cloudDataDir,
+      Optional<String> dataEndpoint,
+      Optional<String> adminEndpoint) {
+    this.projectId = projectId;
+    this.instanceId = instanceId;
+    this.dataflowRegion = dataflowRegion;
+    this.workdir = workdir;
+    this.cloudDataDir = cloudDataDir;
+    this.dataEndpoint = dataEndpoint;
+    this.adminEndpoint = adminEndpoint;
+  }
+
+  public String getProjectId() {
+    return projectId;
+  }
+
+  public String getInstanceId() {
+    return instanceId;
+  }
+
+  public String getDataflowRegion() {
+    return dataflowRegion;
+  }
+
+  /**
+   * Directory that will store ephemeral data that can be re-populated but the test
+   *
+   * <p>Expected Structure:
+   *
+   * <ul>
+   *   <li>staging - dataflow staged jars
+   *   <li>workdir-${uuid} - temporary workdir directory for the a test
+   */
+  public String getWorkdir() {
+    return workdir;
+  }
+
+  public String getDataflowStagingDir() {
+    return ensureTrailingSlash(getWorkdir()) + "staging/";
+  }
+
+  public String getTestWorkdir(UUID testId) {
+    return ensureTrailingSlash(getWorkdir()) + "workdir-" + testId + "/";
+  }
+
+  // TODO: make this ephemeral
+  public String getCloudDataDir() {
+    return cloudDataDir;
+  }
+
+  public Optional<String> getDataEndpoint() {
+    return dataEndpoint;
+  }
+
+  public Optional<String> getAdminEndpoint() {
+    return adminEndpoint;
+  }
+
+  /** Contains persistent fixture data */
+  public static TestProperties fromSystem() {
+    // TODO: once all of the kokoro configs are updated replace this with
+    // getProp("dataflow.work-dir")
+    String workDirKey = "google.dataflow.work-dir";
+    String workDir = System.getProperty(workDirKey);
+    // backwards compat for old setup
+    if (workDir == null) {
+      workDir = System.getProperty("google.dataflow.stagingLocation");
+    }
+    Preconditions.checkNotNull(workDir, "The system property " + workDirKey + " must be specified");
+
+    return new TestProperties(
+        getProp("google.bigtable.project.id"),
+        getProp("google.bigtable.instance.id"),
+        getProp("region"),
+        ensureTrailingSlash(workDir),
+        ensureTrailingSlash(getProp("cloud.test.data.folder")),
+        Optional.ofNullable(System.getProperty("google.bigtable.endpoint.host")),
+        Optional.ofNullable(System.getProperty("google.bigtable.admin.endpoint.host")));
```

Comment:
BIGTABLE_ADMIN_HOST_KEY?  same for data endpoint

Reply:
switched both to use constants